### DISCUSSION TITLE
Numba support?

### DISCUSSION THREAD

**Discussion by fakuivan** [francof2a/fxpmath#75] [New features]
Is it possible to integrate Fxp into Numba? Maybe using something like the [`jitclass`](https://numba.pydata.org/numba-doc/latest/user/jitclass.html) decorator

**Comment by francof2a** [comment]
Interesting question!
I think the short answer is **No**. I did some brief test and Fxp does not support `jitclass` decorator. There are a copule of things that Fxp has implemented in order to increase flexibility that wouldn't allow Numba support, but I think it could be a nice feature to implement in the future.
As this functionality was not planned until now, let me move this issue into a **discussion**.

**Comment by fakuivan** [comment]
The main reason I want to use numba with fxp is that it seems like a convenient way to simulate adaptive filters (that work with the gradiant descend methods) that will need to be run on DSPs. While most DSPs nowadays support floating point math, it is sometimes prefered to just use fixed point whenever possible. The idea would be to simlulate and contrast the performance of the algorithms implemented with fxp vs floats.

**Comment by fakuivan** [comment]
The main reason I want to use numba with fxp is that it seems like a convenient way to simulate adaptive filters (that work with the gradiant descend methods) that will need to be run on DSPs. While most DSPs nowadays support floating point math, it is sometimes prefered to just use fixed point whenever possible. The idea would be to simlulate and contrast the performance of the algorithms implemented with fxp vs floats.

**Comment by fakuivan** [comment]
The main reason I want to use numba with fxp is that it seems like a convenient way to simulate adaptive filters (that work with the gradiant descend methods) that will need to be run on DSPs. While most DSPs nowadays support floating point math, it is sometimes prefered to just use fixed point whenever possible. The idea would be to simlulate and contrast the performance of the algorithms implemented with fxp vs floats.
